skimage/filters/rank/generic.py

```diff
@@ -126,11 +192,51 @@ def _apply_scalar_per_pixel(func, image, selem, out, mask, shift_x, shift_y,
 
 def _apply_vector_per_pixel(func, image, selem, out, mask, shift_x, shift_y,
                             out_dtype=None, pixel_size=1):
+    """
+
+    Parameters
+    ----------
+    func : function
+        Cython function to apply.
+    image : 2-D array (integer, float or boolean)
+        Input image.
+    selem : 2-D array (integer, float or boolean)
+        The neighborhood expressed as a 2-D array of 1's and 0's.
+    out : 2-D array (integer, float or boolean)
+        If None, a new array is allocated.
+    mask : ndarray (integer, float or boolean)
+        Mask array that defines (>0) area of the image included in the local
+        neighborhood. If None, the complete image is used (default).
+    shift_x, shift_y : int
+        Offset added to the structuring element center point. Shift is bounded
+        to the structuring element sizes (center must be inside the given
+        structuring element).
+    out_dtype : data-type, optional
+        Desired output data-type. Default is None, which means we cast output
+        in input dtype.
+    pixel_size : int
+        Dimension of each pixel. Default value is 1.
```

Comment:
```suggestion
        Dimension of each pixel.
```
We don't have to specify default values for pure python functions if no precision is needed.